### PR TITLE
Add `kedro package` as a CI check

### PR DESCRIPTION
**Pull request by ElenaKhaustova**
## Motivation and Context
Solves https://github.com/kedro-org/kedro/issues/3209

## How has this been tested?
1. `make install-test-requirements`
2. `behave` - to run all tests or `behave -n <Scenario>` to run specific test

## Checklist

- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Assigned myself to the PR
- [ ] Added tests to cover my changes

### CODE REVIEW

**Comment by ankatiyar** [comment]
Do you mind also adding this in the github actions workflow - https://github.com/kedro-org/kedro-starters/blob/main/.github/workflows/all-checks.yml so it runs in the CI

**Comment by ElenaKhaustova** [comment]
> Do you mind also adding this in the github actions workflow - https://github.com/kedro-org/kedro-starters/blob/main/.github/workflows/all-checks.yml so it runs in the CI

ah, yes, sure

File: .github/workflows/all-checks.yml [hunk 1]
```diff
@@ -78,3 +78,39 @@ jobs:
...
+      - name: Setup Hadoop binary
+        if: matrix.os == 'windows-latest'
+        run: |
+          Invoke-WebRequest "https://github.com/steveloughran/winutils/blob/master/hadoop-2.7.1/bin/winutils.exe?raw=true" -OutFile winutils.exe
+          Invoke-WebRequest "https://github.com/steveloughran/winutils/blob/master/hadoop-2.7.1/bin/hadoop.dll?raw=true" -OutFile hadoop.dll
+          Move-Item .\hadoop.dll C:\Windows\System32
+          New-Item -ItemType directory -Path C:\hadoop\bin
+          Move-Item .\winutils.exe C:\hadoop\bin
+          echo "HADOOP_HOME=C:\hadoop" | Out-File -Append -Encoding ascii -FilePath $env:GITHUB_ENV
+          echo "PATH=$env:HADOOP_HOME\bin;$env:PATH" | Out-File -Append -Encoding ascii -FilePath $env:GITHUB_ENV
```
Comment (noklam) [inline .github/workflows/all-checks.yml:114]: Do we need this for starter? I am actually surprised we bring back Hadoop setup into `kedro` @ankatiyar, why do we need this? As I understand we don't have dataset settings, so I don't expect we need hadoop other than `kedro-datasets`

Comment (ElenaKhaustova) [reply .github/workflows/all-checks.yml:114]: Without Hadoop setup spark starters were failing on window: https://github.com/kedro-org/kedro-starters/actions/runs/10705672782/job/29685497693

Comment (ankatiyar) [reply .github/workflows/all-checks.yml:114]: Yeah for the pyspark starters ^

**Comment by astrojuanlu** [comment]
Wondering if there's an easier way to setup Hadoop on Windows, maybe through conda (or better, [micromamba](https://mamba.readthedocs.io/en/latest/user_guide/micromamba.html))? According to https://github.com/conda-forge/pyspark-feedstock/issues/34#issuecomment-1193931978, `-c conda-forge pyspark` does not depend on Hadoop, and when I did tests with @ravi-kumar-pilla , an `environment.yml` like this was enough:

```yaml
name: kedro-pyspark
channels:
- conda-forge
dependencies:
- findspark
- openjdk
- pyspark
- python=3.11
```

The fact that you managed to get Hadoop.exe installed through PowerShell is truly impressive, but I doubt Windows users would go that far, and therefore it would maybe be more useful if the way we do things in our CI is representative of how _users_ get to install Kedro.

(Not to mention that I highly doubt _anybody_ is trying to use Kedro + PySpark on Windows natively, but that's another story)

**Comment by ElenaKhaustova** [comment]
> (I don't intend to block the PR over this though, it was more of an informative comment - "if it works, don't touch it")

Since we use that approach to run other e2e tests as well, maybe we should address this in a separate PR?
https://github.com/kedro-org/kedro-starters/blob/dc60fc3696a1b337b3d094de5e8618591f8c070e/.github/workflows/all-checks.yml#L27

**Review by ankatiyar — APPROVED** [review]
LGTM!
I think starters CI needs an update anyway (also need to update actions versions) we can do it separately.

**Comment by ElenaKhaustova** [comment]
@astrojuanlu, @ankatiyar - separate issue opened for updating the whole CI for starters

**Review by merelcht — APPROVED** [review]
LGTM! 👍

**Review by noklam — APPROVED** [review]

**Pull request merged**